server/plugin/virtual_agent.go

```diff
@@ -246,15 +278,42 @@ func (p *Plugin) ProcessResponse(data []byte) error {
 					return err
 				}
 			}
-		//TODO: Modify later to display a proper card.
 		case *OutputCard:
-			var data OutputCardData
-			if err = json.Unmarshal([]byte(res.Data), &data); err != nil {
-				return err
-			}
+			switch res.TemplateName {
+			case OutputCardSmallImageType, OutputCardLargeImageType:
+				var data OutputCardImageData
+				if err = json.Unmarshal([]byte(res.Data), &data); err != nil {
+					return err
+				}
 
-			if _, err = p.DMWithAttachments(userID, p.CreateOutputCardAttachment(&data)); err != nil {
-				return err
+				if _, err = p.DMWithAttachments(userID, p.CreateOutputCardImageAttachment(&data)); err != nil {
+					return err
+				}
+			case OutputCardVideoType:
+				var data OutputCardVideoData
+				if err = json.Unmarshal([]byte(res.Data), &data); err != nil {
+					return err
+				}
+
+				if _, err = p.DMWithAttachments(userID, p.CreateOutputCardVideoAttachment(&data)); err != nil {
+					return err
+				}
+
+				URL := url.URL{RawQuery: data.Link}
```

Comment:
```suggestion
				videoURL := url.URL{RawQuery: data.Link}
```
Or something else that suits better.